Mathlib/NumberTheory/ModularForms/JacobiTheta/TwoVariable.lean

```diff
@@ -14,119 +15,501 @@ This file defines the two-variable Jacobi theta function
 $$\theta(z, \tau) = \sum_{n \in \mathbb{Z}} \exp (2 i \pi n z + i \pi n ^ 2 \tau),$$
 
 and proves the functional equation relating the values at `(z, τ)` and `(z / τ, -1 / τ)`,
-using Poisson's summation formula. We also show holomorphy (in both variables individually,
-not jointly).
+using Poisson's summation formula. We also show holomorphy (jointly in both variables).
+
+Additionally, we show some analogous results about the derivative (in the `z`-variable)
+
+$$\theta'(z, τ) = \sum_{n \in \mathbb{Z}} 2 \pi i n \exp (2 i \pi n z + i \pi n ^ 2 \tau).$$
+
+(Note that the Mellin transform of `θ` will give us functional equations for `L`-functions
+of even Dirichlet characters, and that of `θ'` will do the same for odd Dirichlet characters.)
 -/
 
 open Complex Real Asymptotics Filter Topology
 
-open scoped Real BigOperators
+open scoped BigOperators ComplexConjugate
 
-/-- The two-variable Jacobi theta function,
-`θ z τ = ∑' (n : ℤ), cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)`.
+noncomputable section
 
-The sum is only convergent for `0 < im τ`; we are implictly extending it by 0 for other values of
-`τ`. -/
-noncomputable def jacobiTheta₂ (z τ : ℂ) : ℂ :=
-  ∑' n : ℤ, cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)
+section term_defs
+/-!
+## Definitions of the summands
+-/
 
-/-- A uniform bound for the summands in the Jacobi theta function on compact subsets. -/
-lemma jacobiTheta₂_term_bound {S T : ℝ} (hT : 0 < T) {z τ : ℂ}
-    (hz : |im z| ≤ S) (hτ : T ≤ im τ) (n : ℤ) :
-    ‖cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)‖ ≤ Real.exp (-π * (T * n ^ 2 - 2 * S * |n|)) := by
-  rw [Complex.norm_eq_abs, Complex.abs_exp, (by push_cast; ring :
+/-- Summand in the series for the Jacobi theta function. -/
+def jacobiTheta₂_term (n : ℤ) (z τ : ℂ) : ℂ := cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)
+
+/-- Summand in the series for the Fréchet derivative of the Jacobi theta function. -/
+def jacobiTheta₂_term_fderiv (n : ℤ) (z τ : ℂ) : ℂ × ℂ →L[ℂ] ℂ :=
+  (cexp (2 * π * I * n * z + π * I * n ^ 2 * τ) •
+  ((2 * π * I * n) • (ContinuousLinearMap.fst ℂ ℂ ℂ) +
+  (π * I * n ^ 2) • (ContinuousLinearMap.snd ℂ ℂ ℂ)))
```

Comment:
```suggestion
  cexp (2 * π * I * n * z + π * I * n ^ 2 * τ) •
    ((2 * π * I * n) • (ContinuousLinearMap.fst ℂ ℂ ℂ) +
      (π * I * n ^ 2) • (ContinuousLinearMap.snd ℂ ℂ ℂ))
```
Not sure what the style guide says, but I find this more readable (and the outer parens are not necessary).